lecture/views.py

```
@@ -1,49 +1,278 @@
+
+        # Serve the HLS segment as a binary file response
+        response = FileResponse(open(segment_path, 'rb'), content_type=content_type)
+        response['Content-Disposition'] = f'inline; filename="{smart_str(safe_segment_name)}"'
+        return response
+    
+    except LectureVideo.DoesNotExist:
+        return HttpResponse("Video not found", status=404)
+    except FileNotFoundError:
+        return HttpResponse("Segment file not found", status=404)
+    except ValueError as e:
+        return HttpResponse(f"Error: {str(e)}", status=500)
```

Ruff (0.8.2)

Use explicit conversion flag

Replace with conversion flag

(RUF010)